string tag
float32 probability